pipeline_dp/dp_computations.py

```diff
@@ -529,6 +536,44 @@ def describe(self) -> str:
                 f"l2_sensitivity={self.sensitivity}")
 
 
+class MeanMechanism:
+    """Computes DP mean.
+
+    It computes DP mean as a ratio of DP sum and DP count. For improving
+    utility the normalization to [min_value, max_value] is performed.
+    It works in the following way:
+    1. normalized_sum = \sum(x_i-mid), where mid = (min_value+max_value)/2.
+    2. dp_normalized_sum, dp_count are computed by adding Laplace or Gaussian
+      noise.
+    3. dp_mean = dp_normalized_sum/dp_count + mid.
+
+    This normalization has benefits that normalized_sum has sensitivity
+    (max_value-min_value)/2 which is smaller that
+     sum_sensitivity = max(|min_value|, |max_value|).
+    """
+
+    def __init__(self, range_middle: float, count_mechanism: AdditiveMechanism,
+                 sum_mechanism: AdditiveMechanism):
+        self._range_middle = range_middle
+        self._count_mechanism = count_mechanism
+        self._sum_mechanism = sum_mechanism
+
+    def compute_mean(self, count: int, normalized_sum: float):
+        dp_count = self._count_mechanism.add_noise(count)
+        denominator = max(1.0, dp_count)  # to avoid division on a small number.
```

Comment:
I think it's just to avoid division by 0, no?

Reply:
I think that the problem is broader than division on 0, e.g. with division by 1e-10 we will have a huge number